onnxruntime/core/providers/cpu/nn/instance_norm_helper.h

```diff
@@ -13,8 +14,12 @@ namespace onnxruntime {
 
 class InstanceNormHelper {
  public:
-  static common::Status ValidateInputs(const Tensor* input, const Tensor* scale, const Tensor* B) {
-    if (input->Shape().NumDimensions() < 3) {
+  static common::Status ValidateInputs(const Tensor* input, const Tensor* scale, const Tensor* B, bool is_nhwc = false) {
+    const auto rank = input->Shape().NumDimensions();
+    auto in_dims = input->Shape().GetDims();
+    auto in_channels = is_nhwc ? in_dims[rank - 1] : in_dims[1];
```

Comment:
Nit: Maybe we can defer reading the contents of `in_dims` after we have finished rank validation below ?